### DISCUSSION TITLE
Integrating pipeline with remote repositories

### DISCUSSION THREAD

**Discussion by cberg-aot** [bcgov/quickstart-openshift#991] [Q&A]
Setup:  Need to deploy a support application as part of our solution.  
- Application is opensource, and contains multiple services integrated together.  
- Images for the application services are open-source and hosted on dockerhub.

Problem:  Do not control remote image registry, do not desire to build images.
- Want to integrate(couple) deployment of remote application with current stack (since solution is dependent).
- Deploy & update external microservices (Should be rare) to **Dev, Test & Prod**
- Cannot 'tag' remote image repo 

Question: Is integrating a remote application deployment (which may not be a react application) a good idea?  
Should we extend the design to accommodate this?  
Separate the pipelines & deployments entirely? (I like this idea, but its an anti-pattern to ci/cd testing convergence.  Plus, A depends on B)

**Comment by DerekRoberts** [comment]
We can already consume remote registries, but only public ones.  It's handled in the OpenShift templates, with the QuickStart's example using `-p PROMOTE=docker.io/thing`.

Tagging is handled by a 3rd party action, which we can dig into.  The maintainer appears responsive.
https://github.com/shrink/actions-docker-registry-tag

It'd take some shuffling, but this request seems doable!  Keep the discussion going and let's get a ticket/issue or other plan in place.  :)
